src/box/iproto.cc

```diff
@@ -3298,6 +3298,41 @@ iproto_set_msg_max(int new_iproto_msg_max)
 	}
 }
 
+int
+iproto_session_send(struct session *session,
+		    const char *header, const char *header_end,
+		    const char *body, const char *body_end)
+{
+	struct iproto_connection *con =
+		(struct iproto_connection *)session->meta.connection;
+	if (con->state != IPROTO_CONNECTION_ALIVE) {
+		diag_set(ClientError, ER_SESSION_CLOSED);
+		return -1;
+	}
+
+	ptrdiff_t header_size = header_end - header;
+	ptrdiff_t body_size = body_end - body;
+	ptrdiff_t packet_size = 5 + header_size + body_size;
+	char *buf = (char *)obuf_alloc(con->tx.p_obuf, packet_size);
+	if (buf == NULL) {
+		diag_set(OutOfMemory, packet_size, "obuf_alloc", "buf");
+		return -1;
+	}
+	char *p = buf;
+	*(p++) = INT8_C(0xce);
+	p = mp_store_u32(p, packet_size - 5);
```

Comment:
Why do you always send uint32?

Reply:
Please check this discussion https://github.com/tarantool/tarantool/pull/7994#discussion_r1039474661.